adapters/repos/db/vector/hnsw/search.go

```diff
@@ -269,12 +356,11 @@ func (h *hnsw) searchLayerByVectorWithDistancer(queryVector []float32,
 					h.handleDeletedNode(e.DocID)
 					continue
 				} else {
-					if err != nil {
-						h.pools.visitedListsLock.RLock()
-						h.pools.visitedLists.Return(visited)
-						h.pools.visitedListsLock.RUnlock()
-						return nil, errors.Wrap(err, "calculate distance between candidate and query")
-					}
+					h.pools.visitedListsLock.RLock()
```

Comment:
Previously we checked `if err != nil` before returning the visitedLists. Is this intentional?

Reply:
We still check if err != nil, check current line nr. 353